### DISCUSSION TITLE
Coordinate Not Found

### DISCUSSION THREAD

**Discussion by MargaridaS** [SciTools/iris#5241] [Q&A] [answered]
Dear  SciTools iris team,

I contact you to ask about an error when reading the coordinate 'latitude'. When I load the file `msftmz_Omon_FGOALS-f3-L_historical_r1i1p1f1_gn_185001-201412.nc` and read the coordinate 'latitude' with `ds[0].coord('latitude', dim_coords=False)` the next error appears:
```
import iris
from iris.experimental.equalise_cubes import equalise_attributes
ds = iris.load(['msftmz_Omon_FGOALS-f3-L_historical_r1i1p1f1_gn_185001-201412.nc-ts0'])
equalise_attributes(ds)
ds[0].coord('latitude', dim_coords=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/gpfs/projects/bsc32/software/suselinux/11/software/Iris/2.4.0-
foss-2019b-Python-3.7.4/lib/python3.7/site-packages/scitools_iris-2.4.0-py3.7.egg/iris/cube.py", line 1497, in coord
    raise iris.exceptions.CoordinateNotFoundError(msg)
iris.exceptions.CoordinateNotFoundError: 'Expected to find exactly 1 latitude coordinate, but found none.'
```
However the file already has the coordinate 'latitude':
```
ncdump -h msftmz_Omon_FGOALS-f3-L_historical_r1i1p1f1_gn_185001-201412.nc-ts0
netcdf msftmz_Omon_FGOALS-f3-L_historical_r1i1p1f1_gn_185001-201412 {
dimensions:
	lat = 316 ;
	bnds = 2 ;
	lev = 31 ;
	time = UNLIMITED ; // (1 currently)
	basin = 2 ;
	strlen = 21 ;
variables:
	double lat(lat) ;
		lat:bounds = "lat_bnds" ;
		lat:units = "degrees_north" ;
		lat:axis = "Y" ;
		lat:standard_name = "latitude" ;
		lat:long_name = "Latitude" ;
	double lat_bnds(lat, bnds) ;
	double lev(lev) ;
		lev:bounds = "lev_bnds" ;
		lev:units = "m" ;
		lev:axis = "Z" ;
		lev:positive = "down" ;
		lev:long_name = "ocean depth coordinate" ;
		lev:standard_name = "depth" ;
	double lev_bnds(lev, bnds) ;
	float msftmz(time, basin, lev, lat) ;
		msftmz:standard_name = "ocean_meridional_overturning_mass_streamfunction" ;
		msftmz:long_name = "Ocean Meridional Overturning Mass Streamfunction" ;
		msftmz:comment = "Overturning mass streamfunction arising from all advective mass transport processes, resolved and parameterized." ;
		msftmz:units = "kg s-1" ;
		msftmz:cell_methods = "longitude: sum (comment: basin sum [along zig-zag grid path]) depth: sum time: mean" ;
		msftmz:history = "2019-10-07T01:54:22Z altered by CMOR: replaced missing value flag (1e+35) with standard missing value (1e+20). 2019-10-07T02:15:30Z altered by CMOR: Converted type from \'d\' to \'f\'. 2019-10-07T02:15:30Z altered by CMOR: Inverted axis: lat." ;
		msftmz:missing_value = 1.e+20f ;
		msftmz:_FillValue = 1.e+20f ;
		msftmz:coordinates = "sector" ;
	char sector(basin, strlen) ;
		sector:long_name = "ocean basin" ;
		sector:standard_name = "region" ;
	double time(time) ;
		time:bounds = "time_bnds" ;
		time:units = "days since 1850-01-01 00:00:00" ;
		time:calendar = "365_day" ;
		time:axis = "T" ;
		time:long_name = "time" ;
		time:standard_name = "time" ;
	double time_bnds(time, bnds) ;
```

Do you know why is this error appearing?
I attach the file: [msftmz_Omon_FGOALS-f3-L_historical_r1i1p1f1_gn_185001-201412.zip](https://github.com/SciTools/iris/files/11221437/msftmz_Omon_FGOALS-f3-L_historical_r1i1p1f1_gn_185001-201412.zip)
Thank you very much in advance.
Best regards,
MargaridaS

**Comment by trexfeathers** [comment, marked as answer]
Hi @MargaridaS, sorry for the delay - things have been a bit hectic lately.

From your `ncdump` output we can see that the `latitude` coordinate's variable name is identical to the name of its dimension - `lat(lat)`. Under the [CF conventions](https://cfconventions.org/cf-conventions/cf-conventions.html) this makes `latitude` a **dimension coordinate**, and Iris recognises this.

By specifying `dim_coords=False` in your coordinate search, you are filtering out all dimension coordinates from the search result, including `latitude`. You should either use:

- `ds[0].coord('latitude')`
- `ds[0].coord('latitude', dim_coords=True)`
  This isn't necessary in this case because you know there is only one coordinate called `latitude`.

**Reply by MargaridaS** [reply]
Hi @trexfeathers 

Thank you very much for the information. Now it works properly, thanks a lot!

Best,
MargaridaS
